packages/core/ios/RNEmbrace/SpanRepository.swift

```diff
@@ -50,12 +50,12 @@ class SpanRepository {
     func spanStarted(span: Span) -> String {
         let key = getKey(span)
 
-        if activeSpans.count > MAX_STORED_SPANS {
```

Comment:
@ArielDemarco FYI we were hitting exceptions randomly when running tests on this line complaining that `count` couldn't be retrieved due to some `EXC_BAD_ACCESS` issue, we talked about this briefly in https://github.com/embrace-io/embrace-react-native-sdk/pull/59#discussion_r1695284704 but it does seem like this needed to be done in a thread-safe way, just wanted to call it out in case we were doing anything similar in the ios sdk itself

Reply:
It's safer to do it in the queue for sure! Probably that will fix the bad access exception